tests/components/deconz/test_lock.py

```diff
@@ -0,0 +1,96 @@
+"""deCONZ lock platform tests."""
+from copy import deepcopy
+
+from homeassistant.components import deconz
+import homeassistant.components.lock as lock
+from homeassistant.const import STATE_LOCKED, STATE_UNLOCKED
+from homeassistant.setup import async_setup_component
+
+from .test_gateway import DECONZ_WEB_REQUEST, setup_deconz_integration
+
+from tests.async_mock import patch
+
+LOCKS = {
+    "1": {
+        "etag": "5c2ec06cde4bd654aef3a555fcd8ad12",
+        "hascolor": False,
+        "lastannounced": None,
+        "lastseen": "2020-08-22T15:29:03Z",
+        "manufacturername": "Danalock",
+        "modelid": "V3-BTZB",
+        "name": "Door lock",
+        "state": {"alert": "none", "on": False, "reachable": True},
+        "swversion": "19042019",
+        "type": "Door Lock",
+        "uniqueid": "00:00:00:00:00:00:00:00-00",
+    }
+}
+
+
+async def test_platform_manually_configured(hass):
+    """Test that we do not discover anything or try to set up a gateway."""
+    assert (
+        await async_setup_component(
+            hass, lock.DOMAIN, {"lock": {"platform": deconz.DOMAIN}}
+        )
+        is True
+    )
+    assert deconz.DOMAIN not in hass.data
+
+
+async def test_no_locks(hass):
+    """Test that no lock entities are created."""
+    gateway = await setup_deconz_integration(hass)
+    assert len(gateway.deconz_ids) == 0
```

Comment:
It's good to avoid asserting integration details in tests of integrations.

Reply:
I will create a new PR to remove this from all deconz tests

Reply:
New PR https://github.com/home-assistant/core/pull/41015